### ISSUE TITLE
glue-0.1.0.0 does not compile with GHC 7.10.1

### ISSUE THREAD

**Issue by peti** [seanparsons/glue#1]
Citing from http://hydra.cryp.to/build/945325/log/raw:

```
example/Glue/Example/BatcherExample.hs:33:7:
    Non type-variable argument
      in the constraint: Control.Monad.Base.MonadBase IO m
    (Use FlexibleContexts to permit this)
    When checking that ‘service’ has the inferred type
      service :: forall (m :: * -> *) v.
                 (Eq v, Num v, hashable-1.2.3.2:Data.Hashable.Class.Hashable v,
                  Control.Monad.Base.MonadBase IO m) =>
                 S.HashSet v -> m (M.HashMap v v)
    In the expression:
      do { counter <- newIORef 0;
           let listOfNums = ...;
           let requests = ...;
           let service request = ...;
           .... }
    In an equation for ‘runTest’:
        runTest store name shouldBatch
          = do { counter <- newIORef 0;
                 let listOfNums = ...;
                 let requests = ...;
                 .... }
```

**Comment by seanparsons** [comment]
Yeah, it's a 7.10.1 issue: https://travis-ci.org/seanparsons/glue/jobs/63370410

**Comment by bergmark** [comment]
It also does not compile with GHC 7.4:

```
src/Glue/Failover.hs:36:101:
    Ambiguous occurrence `catch'
    It could refer to either `Prelude.catch',
                             imported from `Prelude' at src/Glue/Failover.hs:5:8-20
                             (and originally defined in `System.IO.Error')
                          or `Control.Exception.Lifted.catch',
                             imported from `Control.Exception.Lifted' at src/Glue/Failover.hs:14:1-45
```

If you care about backwards compatibility you need to hide the import from Prelude or qualifiy the import of `Control.Exception.Lifted.catch'`. If not, setting a `base >= 4.6` bound is sufficient.

**Comment by bergmark** [comment]
Forgot to mention, I revised this on hackage: http://hackage.haskell.org/package/glue-0.1.0.0/revisions/

**Comment by seanparsons** [comment]
@bergmark Ah yeah, I think a 7.4 release is not going to happen, thanks for that.

**Comment by seanparsons** [comment]
Fixed now with the 0.1.1.0 release.
